spanner/client.go

```diff
@@ -150,7 +150,6 @@ func NewClientWithConfig(ctx context.Context, database string, config ClientConf
 	}
 
 	// Prepare gRPC channels.
-	configuredNumChannels := config.NumChannels
```

Comment:
I don't think you should remove this variable. It is used to detect whether the user has set a value for `config.NumChannels`. If not, this value will contain 0 (zero) and in that case, we should ignore any difference with any value that might have been set using `WithGRPCConnectionPool`.

Reply:
I rename `configuredNumChannels` to `hasNumChannelsConfig`, does it sound good?

Reply:
Yes, that sounds good.